core/src/main/kotlin/entity/interaction/Interaction.kt

```diff
@@ -53,20 +74,56 @@ class Interaction(val data: InteractionData, override val applicationId: Snowfla
 
 }
 
+/**
+ * The root command in the interaction.
+ */
 class Command(val data: ApplicationCommandInteractionData) : Entity {
     override val id: Snowflake
         get() = data.id
-    val name = data.name
 
-    val parameters = data.options.orEmpty().map { Parameter(it) }
+    val name get() = data.name
+
+    val options
+        get(): Map<String, OptionValue<*>> = data.options.orEmpty()
+            .filter { it.value !is Optional.Missing<*> }
+            .associate { it.name to it.value.value!! }
+
+    val groups: Map<String, Group>
+        get() = data.options.orEmpty()
+            .filter { it.subCommand.orEmpty().isNotEmpty() }
+            .associate { it.name to Group(it) }
+
+
+    val subCommands: Map<String, SubCommand>
```

Comment:
Should clarify that these are the direct subcommands, not including those in `groups`.